src/tests.py

```diff
@@ -236,13 +238,127 @@ def _integration_test(target_directory):
             assert str(e) == "Branch does not have a parent: master"
 
 
+
```

Comment:
nit: There should be 2 blank lines here instead of 3, according to PEP 8 (eg delete one of these blank lines)